### ISSUE TITLE
Add feature -> Edge detection and crop

### ISSUE THREAD

**Issue by Aman-zishan** [Aman-zishan/DocScanner#13]
AS of now the app simply converts the captured image to pdf without scaling or cropping. It would be better to add a crop option and edge detection feature while scanning documents.

- [ ] edge detection after image is captured

- [x] crop option incase if edge detection fails to get the required outcome

Probably we will need to add a new screen with the preview of image and then processsing the new cropped or scaled image to generate pdf page.

**Comment by Amitbhave** [comment]
@Aman-zishan do you have anything in mind what should be used for edge detection?
For image crop, I was thinking to use [image_cropper](https://pub.dev/packages/image_cropper) plugin.

**Comment by Aman-zishan** [comment]
> @Aman-zishan do you have anything in mind what should be used for edge detection?
> For image crop, I was thinking to use [image_cropper](https://pub.dev/packages/image_cropper) plugin.

There is [edge detection](https://pub.dev/packages/edge_detection) plugin too, the problem is to split the logic into different processes and implement the idea. As of now the camera takes a pic and goes directly to share and print so we have to create a new preview screen which triggers for this edge detection and crop . This screen should also be triggered when users switch to greyscale or multi image capture/picker mode(which will be implemented later #11 and #10)

**Comment by Amitbhave** [comment]
Sure, can I work on this?

**Comment by Aman-zishan** [comment]
> Sure, can I work on this?

Are you implementing the same logic? Please do share a Screenshot while opening a PR too. Thank you!

**Comment by Amitbhave** [comment]
Will it be okay, if we do edge detection at the time of capturing the image only?

**Comment by Aman-zishan** [comment]
> Will it be okay, if we do edge detection at the time of capturing the image only?

As in whenever the camera is used right, umm that would be ok i guess cause if we make use of crop for images thats uploaded from gallery there isnt a requirement for edge detection. Yeah that would do

**Comment by Amitbhave** [comment]
There is this issue with edge_detection package.
https://github.com/sawankumarbundelkhandi/edge_detection/issues/20

**Comment by Aman-zishan** [comment]
> There is this issue with edge_detection package.
> [sawankumarbundelkhandi/edge_detection#20](https://github.com/sawankumarbundelkhandi/edge_detection/issues/20)

Did you try in a physical device?
 
Have you made a good amount of progress?. Then i guess we can open a new branch and work on this in that branch?. So we both can try and fix 

I have created a new [branch](https://github.com/Aman-zishan/DocScanner/tree/beta-test) maybe you can PR to this branch so that i can look into it.

**Comment by Amitbhave** [comment]
Why do you need new branch? I have forked your repo.. While raising PR, I will attach screenshots as well don't worry.

**Comment by Aman-zishan** [comment]
Oh cool then..i thought you were stuck at that error so we could work on it in that branch

**Comment by sawankumarbundelkhandi** [comment]
@Aman-zishan @Amitbhave Please try using the plugin on a real device. I have never tested the edge_detection plugin on the emulator.

**Comment by Aman-zishan** [comment]
> @Aman-zishan @Amitbhave Please try using the plugin on a real device. I have never tested the edge_detection plugin on the emulator.

Yeah saw a lot of issues with emulators while using this package.

**Comment by Amitbhave** [comment]
@Aman-zishan so should we be using this package or not as it doesn't work on emulator?

**Comment by Aman-zishan** [comment]
> @Aman-zishan so should we be using this package or not as it doesn't work on emulator?

I'm actually using a physical device. Can you use one too

**Comment by Amitbhave** [comment]
@Aman-zishan you can go ahead with this using actual physical device.

**Comment by Aman-zishan** [comment]
?? what happened

**Comment by Amitbhave** [comment]
Actually I prefer to use emulator only for testing rather than using my own device.

**Comment by Aman-zishan** [comment]
No problem

**Comment by Prakhar314** [comment]
Can I work on this? I am thinking of using [this plugin](https://github.com/flutter-clutter/flutter-simple-edge-detection/). It is using opencv using dart:ffi.
The example there has an implementation of edge detection and crop.
Maybe we can add some filters as well, using this.

**Comment by Aman-zishan** [comment]
Yes sure

**Comment by Aman-zishan** [comment]
Hey @Prakhar314 , we have already implemented crop feature would be better if you work on edge detection

**Comment by Prakhar314** [comment]
I'm sorry, I won't be able to work on this any time soon.
